src/hooks/useFilterInputChangeHandler.ts

```diff
@@ -1,50 +1,70 @@
-import { useState } from "react";
+import { useState, useMemo } from "react";
 import debounce from "lodash.debounce";
-import { useLocation, useHistory } from "react-router-dom";
-import { url, queryString } from "utils";
+import { useLocation } from "react-router-dom";
+import { FilterHookParams, FilterHookResult } from "hooks/useStatusesFilter";
+import { useUpdateURLQueryParams } from "hooks/useUpdateURLQueryParams";
+import { queryString } from "utils";
 
 const { parseQueryString } = queryString;
-const { updateUrlQueryParam } = url;
-
-const updateQueryParamWithDebounce = debounce(updateUrlQueryParam, 250);
-
-type InputEvent = React.ChangeEvent<HTMLInputElement>;
 
 /**
- * useFilterInputChangeHandler updates the url query param (urlSearchParam) by the
- * the value user enters into the input field.
- * @param  {string} urlSearchParam the url search param that should update
- * @param {boolean} resetPage update url page param to 0 if true
- * @param {(filterBy: string) => void} sendAnalyticsEvent analytics event that is dispatched on filter change
- * @return {[string, (e: InputEvent) => void]}
+ * Filter input state management hook.
+ * @param {FilterHookParams}
+ * @return {FilterHookResult<string>}
  */
-export const useFilterInputChangeHandler = (
-  urlSearchParam: string,
-  resetPage: boolean,
-  sendAnalyticsEvent: (filterBy: string) => void = () => undefined
-): [string, (e: InputEvent) => void] => {
-  const { pathname, search } = useLocation();
-  const { replace } = useHistory();
-
-  const parsed = parseQueryString(search);
-  const inputValue = (parsed[urlSearchParam] || "").toString();
+export const useFilterInputChangeHandler = ({
+  urlParam,
+  resetPage,
+  sendAnalyticsEvent = () => undefined,
+}: FilterHookParams): FilterHookResult<string> => {
+  const { search } = useLocation();
+  const updateQueryParams = useUpdateURLQueryParams();
+  const updateQueryParamWithDebounce = useMemo(
+    () => debounce(updateQueryParams, 250),
+    [updateQueryParams]
+  );
```

Comment:
Is this still necessary? Afaict we no longer update any values as we type. They only change when the user presses `filter` So a debounce seems unnecessary.

Reply:
this is used on the Patches page input where there is no submit btn